### PR TITLE
Add 0.10.0 changelog

### PR DESCRIPTION
**Pull request by benraskin92**
**What this PR does / why we need it**:

Adds 0.10.0 changelog 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,26 @@
 # Changelog
 
+# 0.10.0
+
+## Features
+
+- **M3Query**: Add multi-zonal capability to remote reads (#1687)
+- **M3Query**: Add debug param to namespace get API for better readability (#1698)
```
Comment (richardartoul) [inline CHANGELOG.md]: This sentence reads a little weird. Maybe `to GET /services/m3db/namespace (or whatever the path is) API`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,26 @@
...
+
+## Performance
+
+- **M3DB**: Add forward index write capability which eases index pressure at block boundaries (#1613)
```
Comment (richardartoul) [inline CHANGELOG.md]: @arnikola Is this documented anywhere? We can we add some O.S.S docs

File: CHANGELOG.md (hunk 1 above)
Comment (richardartoul) [reply CHANGELOG.md]: Also is this documented anywhere?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,26 @@
 # Changelog
 
+# 0.10.0
+
+## Features
+
+- **M3Query**: Add multi-zonal capability to remote reads (#1687)
```
Comment (richardartoul) [inline CHANGELOG.md]: @arnikola Same comment about docs

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,26 @@
...
+
+- **M3DB**: Add forward index write capability which eases index pressure at block boundaries (#1613)
+- **M3Query**: Drop empty series from appearing in output when `keepNaNs` option is set to disabled (#1682, #1684)
+- **M3DB**: Build and release M3DB with Go 1.12 (#1674)
+
+## Bug Fixes
+
+- **M3DB**: Fix a bug where peer bootstrapping would sometimes get stuck for a short period of time due to other M3DB nodes (temporarily) returning more than one block of data for a given time period (#1707)
+- **M3DB**: Fix a bug that would cause multiple orders of magnitude slow-down in peer bootstrapping / node replacements when one of the nodes in the cluster was hard down (#1677)
+- **M3Query**: Fix a bug with parsing Graphite find queries by adding `MatchField` and `MatchNotField` match types, and explicitly makes use of them in graphite queries (#1676)
```
Comment (richardartoul) [inline CHANGELOG.md]: makes -> making

File: CHANGELOG.md (hunk 3 above)
Comment (robskillington) [reply CHANGELOG.md]: Can we make this `Add multi-zone and multi-region configuration for coordinator (#1687)`?

File: CHANGELOG.md (hunk 1 above)
Comment (benraskin92) [reply CHANGELOG.md]: Yes, it is: http://m3db.github.io/m3/operational_guide/namespace_configuration/#viewing-a-namespace

**Pull request merged**